src/check_datapackage/internals.py

```diff
@@ -134,3 +241,43 @@ def _get_full_json_path_from_error(error: ValidationError) -> str:
         if match:
             return f"{error.json_path}.{match.group(1)}"
     return error.json_path
+
+
+def _validation_error_to_schema_errors(error: ValidationError) -> Iterable[SchemaError]:
+    current = [_create_schema_error(error)]
+    if not error.context:
+        return current
+    return chain(
+        current,
+        flat_map(_validation_error_to_schema_errors, error.context),
+    )
+
+
+def _create_schema_errors(errors: list[ValidationError]) -> list[SchemaError]:
+    return list(flat_map(_validation_error_to_schema_errors, errors))
```

Comment:
```suggestion
    return flat_map(_validation_error_to_schema_errors, errors)
```

flat map should already output a list.